matchbox_server/src/signaling.rs

```diff
@@ -452,9 +459,212 @@ mod tests {
             _ = &mut timeout => {}
         }
     }
+    #[tokio::test]
+    async fn match_pair_and_other_alone_room_without_next() {
+        let _ = pretty_env_logger::try_init();
+        let api = api();
+
+        let mut client_a = warp::test::ws()
+            .path("/room_name?next=2")
+            .handshake(api.clone())
+            // .handshake(ws_echo())
```

Comment:
other nit, that PR adds a lot of duplicatd code for tests ; I guess it's usual so why not, but it might be a good time to split in another file for tests ?

Reply:
I feel that in tests it's far better to err on the side of simple rather than dry. imo it's only worth it if the test becomes easier to read and debug.

Reply:
I'm ok with repetition in tests :+1:.

I especially want to be clear that signalling.rs sees its line count going from 460 to 650, so it might be a good time to move tests out of that file.